kernelci/config/__init__.py

```diff
@@ -27,7 +28,36 @@
 import kernelci.config.test
 
 
-def load_yaml(config_path):
+def _iterate_yaml_files(config_path):
+    if config_path.endswith('.yaml'):
+        yaml_files = [config_path]
+    else:
+        yaml_files = glob.glob(os.path.join(config_path, "*.yaml"))
+    for yaml_path in yaml_files:
+        with open(yaml_path) as yaml_file:
+            data = yaml.safe_load(yaml_file)
+            yield(yaml_path, data)
+
+
+def validate_yaml(config_path, entries):
+    for yaml_path, data in _iterate_yaml_files(config_path):
+        for name, value in data.items():
```

Comment:
Maybe we could use a generator expression here instead of for loop with if..continue?
```python
for name, value in (n, v for n, v in data.items() if n in entries):
```

Reply:
Fair enough, with a couple more brackets though: `(n, v) for n, v in data.items`

Reply:
Fixed